bucket_test.go

```diff
@@ -464,6 +464,217 @@ func TestBucket_Delete_NonExisting(t *testing.T) {
 	}
 }
 
+func TestBucket_MoveBucket(t *testing.T) {
+	testCases := []struct {
+		name                    string
+		srcBucketName           string
+		subBucketName           string
+		subBucketKey            string
+		subBucketValue          string
+		dstBucketName           string
+		subBucketExistSrcBucket bool
+		subBucketExistDstBucket bool
+		keyNoSubBucketSrcBucket bool
+		keyNoSubBucketDstBucket bool
+		expErr                  error
+	}{
+		{
+			"happy path",
+			"srcBucket",
+			"subBucket",
+			"this is subBucket key",
+			"this is subBucket value",
+			"dstBucket",
+			true,
+			false,
+			false,
+			false,
+			nil,
+		},
+		{
+			"subBucket not exist in srcBucket",
+			"srcBucket",
+			"subBucket",
+			"this is subBucket key",
+			"this is subBucket value",
+			"dstBucket",
+			false,
+			false,
+			false,
+			false,
+			berrors.ErrBucketNotFound,
+		},
+		{
+			"suBucket exist in dstBucket",
```

Comment:
@ahrtr @fuweid  i added three cases to test when 

- src is RootBucket
- dst is RootBucket
- src & dst are RootBucket

I tried to refactor the test because it is very confusing, but i realised that the helpers i will create will also be confusing

also the booleans i use imho are necessary because of the cases I am trying to cover

ptal and let me know how to imporove

Reply:
I was thinking that we can introduce a helper to build bucket or key/values by path which is like jsonpath. 

```go
type valuePath struct {
        buckets []string
        isBucket bool
        key []byte
        value []byte  // ignored if isBucket is true
}

func initDataset(t *testing.T, db *bbolt.DB, vs []valuePath) {
       ....
}

var source []string // /key1/key2/key/..
var target []string // /keyx/keyy/...
var expected valuePath
var hasError bool
```

IMO, using `source` and `target` is easier to understand input. No need to introduce extra boolean as toggle.
Just my 2 cents.

Reply:
ok i follow this approach now 👍🏽

Reply:
@fuweid @ahrtr  is there example in the test files that can help ?

Reply:
> I was thinking that we can introduce a helper to build bucket or key/values by path which is like jsonpath.
> 
> ```go
> type valuePath struct {
>         buckets []string
>         isBucket bool
>         key []byte
>         value []byte  // ignored if isBucket is true
> }
> 
> func initDataset(t *testing.T, db *bbolt.DB, vs []valuePath) {
>        ....
> }
> 
> var source []string // /key1/key2/key/..
> var target []string // /keyx/keyy/...
> var expected valuePath
> var hasError bool
> ```
> 
> IMO, using `source` and `target` is easier to understand input. No need to introduce extra boolean as toggle. Just my 2 cents.

@fuweid now i am trying to add a case `subBucket does not exist in srcBucket` 

I need to assert for error in this case, also skip the creation of the very last bucket in `srcBucketPath` 

i cant find a way but using `bool` :/ 

please correct me if i am wrong